File: Lab2/app/src/main/java/com/example/lab2/ETextFont.java
package com.example.lab2;

import android.graphics.Typeface;

import java.util.Arrays;

public enum ETextFont {

    MONOSPACE(Typeface.create("monospace", Typeface.NORMAL), "monospace"),
    CURSIVE(Typeface.create("cursive", Typeface.NORMAL), "cursive"),
    CASUAL(Typeface.create("casual", Typeface.NORMAL), "casual"),
    SANS_SERIF(Typeface.create("sans-serif", Typeface.NORMAL), "sans-serif"),
    SANS_SERIF_BLACK(Typeface.create("sans-serif-black", Typeface.NORMAL), "sans-serif-black");

    private final Typeface typeface;
    private final String name;

    ETextFont(Typeface typeface, String name) {
        this.typeface = typeface;
        this.name = name;
    }

    public static ETextFont getTextFontByTypeface(Typeface typeface) {
        return Arrays.stream(ETextFont.values())
                .filter(eTextFont -> eTextFont.getTypeface().equals(typeface))
                .findAny()
                .orElse(null);
    }

    public static ETextFont getTextFontByName(String name) {
        return Arrays.stream(ETextFont.values())
                .filter(eTextFont -> eTextFont.getName().equals(name))
                .findAny()
                .orElse(null);
    }

    public Typeface getTypeface() {
        return typeface;
    }

    public String getName() {
        return name;
    }
}

File: Lab3/app/src/main/java/com/example/lab3/enumerable/ETextFont.kt
package com.example.lab3.enumerable

import android.graphics.Typeface
import java.util.*

enum class ETextFont(val typeface: Typeface, val fontName: String) {

    MONOSPACE(Typeface.create("monospace", Typeface.NORMAL), "monospace"),
    CURSIVE(Typeface.create("cursive", Typeface.NORMAL), "cursive"),
    CASUAL(Typeface.create("casual", Typeface.NORMAL), "casual"),
    SANS_SERIF(Typeface.create("sans-serif", Typeface.NORMAL), "sans-serif"),
    SANS_SERIF_BLACK(Typeface.create("sans-serif-black", Typeface.NORMAL), "sans-serif-black");

    companion object {

        fun getTextFontByTypeface(typeface: Typeface): ETextFont = Arrays.stream(values())
            .filter { e -> e.typeface == typeface }
            .findAny()
            .orElse(null)

        fun getTextFontByName(fontName: String): ETextFont = Arrays.stream(values())
            .filter { e -> e.fontName == fontName }
            .findAny()
            .orElse(null)
    }

}